File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
static int op(int a, int b) {
    return a * b;
}
    bool checkDivisibility(int n) {
       vector<int> v;
       int t=n;
       while(t)
       {
        v.push_back(t%10);
        t/=10;
       } 
       int digs=accumulate(v.begin(),v.end(),0);
       int digp=accumulate(v.begin(),v.end(),1,op);
       return n%(digs+digp)==0?true:false;
    }
};

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.java
class Solution {
    public boolean checkDivisibility(int n) {
        int s=0;
        long p=1l;
        int t=n;
        while(n!=0)
        {
            s+=n%10;
            p*=n%10;
            n/=10;
        }
        return t%(s+p)==0?true:false;

    }
}